usl_pipeline/cloud_functions/main.py

```diff
@@ -150,25 +156,62 @@ def write_study_area_metadata(cloud_event: functions_framework.CloudEvent) -> No
 
 @functions_framework.cloud_event
 @_retry_and_report_errors()
-def write_flood_scenario_metadata(cloud_event: functions_framework.CloudEvent) -> None:
-    """Writes metadata for uploaded flood scenario config.
+def write_flood_scenario_metadata_and_features(
+    cloud_event: functions_framework.CloudEvent,
+) -> None:
+    """Writes metadata and features for uploaded flood scenario config.
 
     This function is triggered when files for rainfall configuration are uploaded to
-    GCS. It writes an entry to the metastore for the configuration.
+    GCS. It writes an entry to the metastore for the configuration and writes the
+    rainfall configuration as a feature vector into the GCS features bucket.
     """
     file_name = pathlib.PurePosixPath(cloud_event.data["name"])
     # Distinguish between Rainfall_Data and CityCat_Config files, the latter of which we
     # don't need to record.
     if file_name.name.startswith("Rainfall_Data_"):
+        storage_client = storage.Client()
         db = firestore.Client()
 
+        config_blob = storage_client.bucket(cloud_event.data["bucket"]).blob(
+            cloud_event.data["name"]
+        )
+        with config_blob.open("rt") as rain_fd:
+            as_vector = _rain_config_as_vector(rain_fd)
+
+        vector_blob = storage_client.bucket(cloud_storage.FEATURE_CHUNKS_BUCKET).blob(
+            f"rainfall/{file_name.with_suffix('.npy')}"
+        )
+        _write_as_npy(vector_blob, as_vector)
+
         metastore.FloodScenarioConfig(
-            gcs_path=f"gs://{cloud_event.data['bucket']}/{cloud_event.data['name']}",
+            gcs_path=f"gs://{config_blob.bucket.name}/{config_blob.name}",
+            as_vector_gcs_path=f"gs://{vector_blob.bucket.name}/{vector_blob.name}",
             # File names should be in the form <parent_config_name>/<file_name>
             parent_config_name=file_name.parent.name,
         ).set(db)
 
 
+def _rain_config_as_vector(rain_fd: TextIO) -> NDArray:
+    """Converts the rainfall config contents into a vector describing the rainfall.
```

Comment:
This function deserves to be placed in usl_lib common logic. There might be other use-cases to parse such files. In general I would vote for extracting any complexities in the logic as common sharable functions.

Reply:
How about usl_lib/readers/rain_config_readers.py?

Reply:
good idea, done.